### PR TITLE
Unbreak non darwin unified builds

### PR DESCRIPTION
**Pull request by davezarzycki**
Fix regression from #742

### CODE REVIEW

**Comment by davezarzycki** [comment]
@swift-ci please smoke test

**Comment by davezarzycki** [comment]
@swift-ci please test platform

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -37,7 +41,11 @@ if(TARGET lldb-mi)
 endif()
 
 if(NOT LLDB_BUILT_STANDALONE)
-  list(APPEND LLDB_TEST_DEPS yaml2obj dsymutil)
+  if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+    list(APPEND LLDB_TEST_DEPS yaml2obj dsymutil)
+  else()
```
Comment (dcci) [inline test/CMakeLists.txt:46]: To the best of my understanding, `yaml2obj` is available anywhere. Any reason why you're limiting this change to Darwin? (Also, yaml2obj is required to run the `lit` tests).

**Review by dcci — CHANGES_REQUESTED** [review]

File: test/CMakeLists.txt (hunk 1 above)
Comment (davezarzycki) [reply test/CMakeLists.txt:46]: If you look closer, this pull request does not change anything about `yaml2obj` or its usage, just `dsymutil`.

**Review by dcci — APPROVED** [review]
My bad. This lgtm.

**Pull request merged**